### PR TITLE
Enable git completion of subcommands

### PR DESCRIPTION
**Pull request by gylaz**

### CODE REVIEW

**Comment by JoelQ** [comment]
Can you explain a little of how this works in the commit message? Maybe also link to some docs?

File: zshrc [hunk 1]
```diff
@@ -57,6 +57,9 @@ bindkey "^Y" accept-and-hold
 bindkey "^N" insert-last-word
 bindkey -s "^T" "^[Isudo ^[A" # "t" for "toughguy"
 
+# git completion
```
Comment (gabebw) [inline zshrc]: Maybe: `Enable git completion for custom subcommands, like git-pr and git-up`

**Comment by jferris** [comment]
Does this also make subcommands complete their arguments? For example, does `git delete-branch` complete branch names?

**Comment by gylaz** [comment]
This does not help with completion of subcommand arguments. Also, our completion of subcommands still doesn't work for our custom `g` function.

**Comment by croaky** [comment]
Completion of subcommands with `git` and `g` is working for me without this change. thoughtbot/dotfiles + https://github.com/croaky/dotfiles

**Comment by gylaz** [comment]
@croaky What version of `git` are you on?

**Comment by geoffharcourt** [comment]
In case it helps, I'm not getting subcommand tab completion on `g` or `git`.

git version 2.4.0
hub version 2.2.0-rc1-g27367ab

**Comment by croaky** [comment]
git version 2.3.5

**Comment by croaky** [comment]
Upgraded to git version 2.4.0 and am still able to tab complete with `g` and `git`.

**Comment by mike-burns** [comment]
This is a simple and easy-to-read addition, but I do highly recommend https://github.com/thoughtbot/gitsh instead.

**Comment by gylaz** [comment]
Closing this in favor of #387.

**Pull request closed without merging**